modal.js

```diff
@@ -0,0 +1,247 @@
+  sourceLink.classList.add('source-link');
+  sourceLink.setAttribute('href', project.sourceLink);
+  sourceLink.textContent = 'Source Link';
+  linkContainer.appendChild(sourceLink);
+  descriptionContainer.appendChild(linkContainer);
+  modalArr.push(modalContainer);
+});
+
+cardArr[0].setAttribute('class', 'card card-1');
+cardArr[1].setAttribute('class', 'card card-2');
+cardArr[2].setAttribute('class', 'card card-3');
+cardArr[3].setAttribute('class', 'card card-4');
+modalArr[0].setAttribute('class', 'modal-container modal-1');
+modalArr[1].setAttribute('class', 'modal-container modal-2');
+modalArr[2].setAttribute('class', 'modal-container modal-3');
+modalArr[3].setAttribute('class', 'modal-container modal-4');
+const modal1 = document.querySelector('.modal-1');
+const modal2 = document.querySelector('.modal-2');
+const modal3 = document.querySelector('.modal-3');
+const modal4 = document.querySelector('.modal-4');
+buttonsArr[0].setAttribute('class', 'button button-1');
+buttonsArr[1].setAttribute('class', 'button button-2');
+buttonsArr[2].setAttribute('class', 'button button-3');
+buttonsArr[3].setAttribute('class', 'button button-4');
+const button1 = document.querySelector('.button-1');
+const button2 = document.querySelector('.button-2');
+const button3 = document.querySelector('.button-3');
+const button4 = document.querySelector('.button-4');
+closeModalArr[0].setAttribute('class', 'close-modal close-modal-1');
+closeModalArr[1].setAttribute('class', 'close-modal close-modal-2');
+closeModalArr[2].setAttribute('class', 'close-modal close-modal-3');
+closeModalArr[3].setAttribute('class', 'close-modal close-modal-4');
+const closeModal1 = document.querySelector('.close-modal-1');
+const closeModal2 = document.querySelector('.close-modal-2');
+const closeModal3 = document.querySelector('.close-modal-3');
+const closeModal4 = document.querySelector('.close-modal-4');
+button1.addEventListener('click', () => {
+  modal1.style.display = 'flex';
+  modal2.style.display = 'none';
+  modal3.style.display = 'none';
+  modal4.style.display = 'none';
+});
+closeModal1.addEventListener('click', () => {
+  modal1.style.display = 'none';
+});
+button2.addEventListener('click', () => {
+  modal1.style.display = 'none';
+  modal3.style.display = 'none';
+  modal4.style.display = 'none';
+  modal2.style.display = 'flex';
+});
+closeModal2.addEventListener('click', () => {
+  modal2.style.display = 'none';
+});
+button3.addEventListener('click', () => {
+  modal1.style.display = 'none';
+  modal2.style.display = 'none';
+  modal4.style.display = 'none';
+  modal3.style.display = 'flex';
+});
+closeModal3.addEventListener('click', () => {
+  modal3.style.display = 'none';
+});
+button4.addEventListener('click', () => {
+  modal1.style.display = 'none';
+  modal2.style.display = 'none';
+  modal3.style.display = 'none';
+  modal4.style.display = 'flex';
+});
+closeModal4.addEventListener('click', () => {
+  modal4.style.display = 'none';
+});
```

Comment:
Optional

- [ ] Reading your code with ease would need you to add spaces after major feature implementations. Adding of single line comments makes it even easier.